Added warning to _env_proxy_settings if passing through cidr to no_proxy

pengale opened the pull request:
_env_proxy_settings will pass the value of juju-no-proxy transparently
to NO_PROXY in a command's environment. juju-no-proxy may contain a
cidr, which is not supported by most applications using NO_PROXY. We
therefore log a warning message that things may break, so that charm
authors can troubleshoot if their charm fails as a result.

Commit:
Added warning to _env_proxy_settings if passing through cidr to no_proxy

_env_proxy_settings will pass the value of juju-no-proxy transparently
to NO_PROXY in a command's environment. juju-no-proxy may contain a
cidr, which is not supported by most applications using NO_PROXY. We
therefore log a warning message that things may break, so that charm
authors can troubleshoot if their charm fails as a result.

Commit:
Added tests for cidr re.

Added a series of simple tests to verify that our regular expression
catches cidrs and not ips.

dshcherb commented on charmhelpers/fetch/ubuntu.py:
```diff
@@ -181,6 +181,11 @@
 CMD_RETRY_COUNT = 3  # Retry a failing fatal command X times.
 
+CONTAINS_CIDR = re.compile(r'.+?[0-9:]+\/[0-9]')
+CIDR_WARNING = ('Passing NO_PROXY string that includes a cidr. '
```
Maybe we should also check for the wildcard syntax with hostnames, see https://bugs.launchpad.net/juju/+bug/1773463

Commit:
Added wildcard domains to check for cidrs in fetch/ubuntu

juju-no-proxy can also take domains as an argument. For example
.foo.com or *.foo.com. Tweaked a check for cidrs so that it catches
these as well.
